lib/transport/selenium-webdriver/browserstack.js

```diff
@@ -46,34 +47,29 @@ class Browserstack extends Selenium {
 
     const {desiredCapabilities} = this.settings;
     //checking for legacy-ways for providing config
-    this.bstackOptions = {
+    defaultsDeep(this.settings.desiredCapabilities['bstackOptions'], {
```

Comment:
we need to ensure that `this.settings.desiredCapabilities['bstackOptions']` is defined first and should it actually be `['bstack:options']`?